### PR TITLE
CIV-10172 Remaining Day calculation

### PR DESCRIPTION
**Pull request by ElifDenizEsea**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-10172

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by kenneth-hmcts — APPROVED** [review]

**Review by portizdev — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/service/citizenui/DashboardClaimInfoService.java [hunk 1]
```diff
@@ -111,6 +111,7 @@ private DashboardClaimInfo translateCaseDataToDashboardClaimInfo(CaseDetails cas
             .defendantName(nonNull(caseData.getRespondent1()) ? caseData.getRespondent1().getPartyName() : null)
             .claimAmount(nonNull(caseData.getTotalClaimAmount()) ? caseData.getTotalClaimAmount() : null)
             .admittedAmount(caseData.getPartAdmitPaidValuePounds())
+            .responseDeadlineTime(caseData.getRespondent1ResponseDeadline())
```
Comment (TatsyMaslova) [inline src/main/java/uk/gov/hmcts/reform/civil/service/citizenui/DashboardClaimInfoService.java:114]: does it work for cmc claims? don't you need to convert to localdatetime from localdate response deadline for cmc?

Comment (TatsyMaslova) [reply src/main/java/uk/gov/hmcts/reform/civil/service/citizenui/DashboardClaimInfoService.java:114]: please have a look at ClaimStoreService line 53

Comment (ElifDenizEsea) [reply src/main/java/uk/gov/hmcts/reform/civil/service/citizenui/DashboardClaimInfoService.java:114]: I added a new field not to impact other flows

Comment (TatsyMaslova) [reply src/main/java/uk/gov/hmcts/reform/civil/service/citizenui/DashboardClaimInfoService.java:114]: ok but dont we need to this calculation for cmc claims too?

**Review by kenneth-hmcts — APPROVED** [review]

**Review by kamilewandowski — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/service/claimstore/ClaimStoreService.java [hunk 2]
```diff
@@ -51,6 +51,10 @@ private List<DashboardClaimInfo> translateCmcClaimToClaimInfo(List<CmcClaim> cmc
                                               .claimantName(cmcClaim.getClaimantName())
                                               .defendantName(cmcClaim.getDefendantName())
                                               .responseDeadline(cmcClaim.getResponseDeadline())
+                                              .responseDeadlineTime(null
```
Comment (TatsyMaslova) [inline src/main/java/uk/gov/hmcts/reform/civil/service/claimstore/ClaimStoreService.java:54]: perhaps use Optional.ofNullable

File: src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/DashboardClaimInfo.java [hunk 3]
```diff
@@ -58,15 +63,10 @@ public class DashboardClaimInfo {
...
-        return Optional.ofNullable(responseDeadline)
-            .filter(deadline ->
-                        deadline.isAfter(LocalDate.now()))
-            .map(deadline ->
-                     LocalDate.now().until(
-                         deadline,
-                         ChronoUnit.DAYS
-                     ))
-            .orElse(0L);
+        if (null != responseDeadlineTime) {
```
Comment (TatsyMaslova) [inline src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/DashboardClaimInfo.java:66]: can you use Optional.ofNullable here?

**Review by kenneth-hmcts — APPROVED** [review]

**Review by kamilewandowski — DISMISSED** [review]

**Review by neeta-hmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/service/citizenui/DashboardClaimInfoService.java (hunk 1 above)
Comment (ElifDenizEsea) [reply src/main/java/uk/gov/hmcts/reform/civil/service/citizenui/DashboardClaimInfoService.java:114]: cmc added

File: src/main/java/uk/gov/hmcts/reform/civil/service/claimstore/ClaimStoreService.java (hunk 2 above)
Comment (ElifDenizEsea) [reply src/main/java/uk/gov/hmcts/reform/civil/service/claimstore/ClaimStoreService.java]: done

File: src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/DashboardClaimInfo.java (hunk 3 above)
Comment (ElifDenizEsea) [reply src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/DashboardClaimInfo.java]: done

**Review by kenneth-hmcts — APPROVED** [review]

**Review by jeswanth-hmcts — APPROVED** [review]

**Review by laperla11 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/service/citizenui/DashboardClaimInfoService.java (hunk 1 above)
Comment (portizdev) [reply src/main/java/uk/gov/hmcts/reform/civil/service/citizenui/DashboardClaimInfoService.java:114]: Resolving the conversation. @TatsyMaslova is advised to check the comment again when she returns.

**Pull request merged**